### ISSUE TITLE
Challenge 40 - Compressing AI Numerical Weather Prediction (NWP) models for edge deployment

### ISSUE THREAD

**Issue by DiegoDomenici** [ECMWFCode4Earth/Challenges_2026#13]
> **Africa Stream**

### Mentors

- Shruti Nath
- Rendani Mbuvha

### Skill Required

- Experience with Python programming
- Expertise in statistical analysis
- Background in Earth Sciences or related fields
- Some experience with Machine Learning desirable
- Some experience with low-precision computing desirable

### Goal

Edge deployment of AI NWP on local, resource-constrained computing units that are at the last mile of data delivery is currently limited by model size and compute required for inference. This challenge invites the exploration of model compression techniques to compress massive AI NWP models to compact versions, whilst not compromising on model accuracy. The smallest most accurate model wins!

### Description of the Challenge

Artificial Intelligence based Numerical Weather Prediction (AI NWP) models are changing the landscape of traditional weather forecasting with current AI NWP models – such as AIFS – achieving comparable skill to physics-based models. However, these models are traditionally "heavy," often requiring GPUs for inference. This creates a significant barrier for last-mile data delivery in operational scenarios where compute power is restricted to standard consumer-grade hardware (laptops).
This challenge tasks participants with exploring techniques of compressing AIFS so as to enable efficient local execution without sacrificing forecast accuracy. Participants will explore model compression techniques—ranging from Structural Pruning and Low-Rank Approximation to Quantization and Self-Distillation (amongst others)—the goal being to bridge the gap between high-end AI research and practical, on-site deployment. The ultimate output will be a truly "laptop-ready" AI NWP model that provides accurate, global inference at a 1 degree resolution and 6-hourly timestep across all available variables of the starting AIFS model. 
The primary metric of success for participants should be magnitude of model size reduction alongside inference latency on a mid-range laptop. Secondarily, the forecast accuracy against the baseline, uncompressed model will also be considered.

Proposed Steps:
-Model setup: Run preliminary tests on the baseline model
-Reduce model parameter redundancy: Explore methods to remove redundant weights within the models
-Speed-up inference through lowered precision: Implement a framework for quantisation/low-precision computing
-Explore distillation methods: see whether knowledge distillation improves accuracy of compressed models
-Final evaluation: publish primary and secondary evaluation metrics

What we can offer:
-A starting, relatively lightweight AIFS model version
-Baseline evaluation of the AIFS model to beat
-Tools for model evaluation and Africa-focussed evaluation criteria

### Evaluation criteria

- Innovative approach
- Easy to maintain / Future-proof approach
- Feasibility
- Proposed potential reduction in model size and inference latency

**Comment by trakasa** [comment]
Dear @LiganiumInc ,
thanks a lot for your interest in the challenge! It would be great seeing a submission by you.
If your application is selected, you can publish a journal paper based on the results of the challenge. This is not foreseen or a must in the programme, but it would be great and we encourage you to do so!

Hope that helps!
Ciao, Athina

**Comment by Khansa435** [comment]
Hi there, 
I'm Khansa, currently a Computer Science undergrad and Iam interested to apply for this project. But I have a few queries:

1)Is there any communication channel(discord, slack etc) to discuss about project details etc and what my proposal should include
2) Also, like other open source programs appreciate or require it necessary to have some prior contributions(PRs made) in the project that are submitting to. Is there any such requiremnt for this project?
3) I have expertise in Python,data analytics and various machine learning frameworks including but not limited to NumPy, pandas, scikit-learn,Tensorflow, PyTorch and writing automation script. Currently Iam also contributing to time series library in Python (i.e sktime). So i do have most of skills required for this project but I just don't have "Background in Earth Sciences or related fields". Would that be okay?

**Comment by EsperanzaCuartero** [comment]
Dear [Khansa435](https://github.com/Khansa435)
Many thanks for your interest in this challenge.

In the challenge description (above), you can find all the requirements for the challenge, along with proposed steps. If you have a specific technical question, please post it here, and a mentor will address it. 

Before writing your submission, I will encourage you to watch the [Q&A webinars from March on the Code for Earth YouTube channel](https://www.youtube.com/@ecmwf-codeforearth) where we explained in detail the questionnaire contained in the submission form and how to better fulfil it. 

We don't require a prior contribution in order to apply for this challenge. 

Although a background in Earth Sciences or related fields is listed as a required skill, it shouldn't discourage you from submitting a proposal. Evaluators will review and rate your submission overall. 

 Please bear in mind that the deadline to submit a proposal is tomorrow, 9 April. 

Best regards
Esperanza

**Comment by Khansa435** [comment]
Thanks a lot @EsperanzaCuartero for clearing my queries and a detailed response on them. But I waited to have them answered in order to start my proposal but received a response a day before April 9th and couldn't submit application in time. Are you still accepting applications or if there is some extension or relaxation in the deadline?, I would be happy to submit an application.

Best, 
Khansa
